pkgs/development/python-modules/hledger-utils/default.nix

```
@@ -0,0 +1,66 @@
+{lib
+, buildPythonPackage
+, fetchFromGitLab
+, setuptools
+, setuptools-scm
+, unittestCheckHook
+, hledger
+, perl
+, rich
+, pandas
+, scipy
+, psutil
+, matplotlib
+, drawilleplot
+, asteval
+}:
+
+buildPythonPackage rec {
+  pname = "hledger-utils";
+  version = "1.12.1";
+
+  format = "pyproject";
+
+  src = fetchFromGitLab {
+    owner = "nobodyinperson";
+    repo = "hledger-utils";
+    rev = "v${version}";
+    hash = "sha256-uAFqBNRET3RaWDTyV53onrBs1fjPR4b5rAvg5lweUN0=";
+  };
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeBuildInputs = [
+    setuptools
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [
+    rich
+    pandas
+    scipy
+    psutil
+    matplotlib
+    drawilleplot
+    asteval
+  ];
+
+  checkInputs = [
+    unittestCheckHook
+  ];
+
+  nativeCheckInputs = [
+    hledger
+    perl
+  ];
+
+  preCheck = "export PATH=$out/bin:$PATH";
```

```suggestion
  preCheck = ''
    export PATH=$out/bin:$PATH";
  '';
```

Simpler to append/modify/remove.